WordPress/src/main/java/org/wordpress/android/ui/mysite/SelectedSiteRepository.kt

```diff
@@ -21,7 +21,12 @@ class SelectedSiteRepository
     private val _selectedSiteChange = MutableLiveData<SiteModel>()
     private val _showSiteIconProgressBar = MutableLiveData<Boolean>()
     val selectedSiteChange = _selectedSiteChange as LiveData<SiteModel>
-    val siteSelected = _selectedSiteChange.map { it?.id }.distinctUntilChanged()
+    val siteSelected: LiveData<Int?> by lazy {
+        val result = MediatorLiveData<Int>()
+        result.addSource(_selectedSiteChange) { site -> result.value = site?.id }
+        result.value = null
+        result.distinctUntilChanged()
+    }
```

Comment:
If emitting `null` as the very first value solves this issue, maybe we could just keep the `siteSelected` logic as it was and have the default value of `_selectedSiteChange` be `null`, like this:

```kotlin
  private val _selectedSiteChange = MutableLiveData<SiteModel?>(null)
  private val _showSiteIconProgressBar = MutableLiveData<Boolean>()
  val selectedSiteChange = _selectedSiteChange as LiveData<SiteModel?>
  val siteSelected = _selectedSiteChange.map { it?.id }.distinctUntilChanged()
```

I've tested the code above just to make sure and, as far as I can see, everything seems to work as expected (including the new test you added). Let me know what you think!